src/commands/database/mysql/create.tsx

```diff
@@ -71,59 +76,85 @@ export class Create extends ExecRenderBaseCommand<typeof Create, Result> {
 
     const password = await this.getPassword(p);
 
-    const db = await p.runStep("creating MySQL database", async () => {
-      const r = await this.apiClient.database.createMysqlDatabase({
-        projectId,
-        data: {
-          database: {
-            projectId,
-            description,
-            version,
-            characterSettings: {
-              collation,
-              characterSet,
-            },
-          },
-          user: {
-            password,
-            externalAccess,
-            accessLevel: accessLevel as "full" | "readonly",
-          },
-        },
-      });
-
-      assertStatus(r, 201);
-      return r.data;
-    });
+    const [db, eventId] = await this.createMySQLDatabase(
+      p,
+      projectId,
+      description,
+      version,
+      collation,
+      characterSet,
+      password,
+      externalAccess,
+      accessLevel,
+    );
 
     const database = await p.runStep("fetching database", async () => {
-      const r = await this.apiClient.database.getMysqlDatabase({
-        mysqlDatabaseId: db.id,
-      });
-      assertStatus(r, 200);
-
-      return r.data;
+      const getWithRetry = withAttemptsToSuccess(
+        this.apiClient.database.getMysqlDatabase,
+      );
+      return (
+        await getWithRetry({
+          mysqlDatabaseId: db.id,
+          headers: { "if-event-reached": eventId },
+        })
+      ).data;
     });
 
     const user = await p.runStep("fetching user", async () => {
-      const r = await this.apiClient.database.getMysqlUser({
-        mysqlUserId: db.userId,
-      });
-      assertStatus(r, 200);
-
-      return r.data;
+      const getWithRetry = withAttemptsToSuccess(
+        this.apiClient.database.getMysqlUser,
+      );
+      return (
+        await getWithRetry({
+          mysqlUserId: db.userId,
+          headers: { "if-event-reached": eventId },
+        })
+      ).data;
     });
 
-    p.complete(
-      <Success>
-        The database <Value>{database.name}</Value> and the user{" "}
-        <Value>{user.name}</Value> were successfully created.
-      </Success>,
-    );
+    await p.complete(<DatabaseCreateSuccess database={database} user={user} />);
 
     return { databaseId: db.id, userId: db.userId };
   }
 
+  private async createMySQLDatabase(
+    p: ProcessRenderer,
+    projectId: string,
+    description: string,
+    version: string,
+    collation: string,
+    characterSet: string,
+    password: string,
+    externalAccess: boolean,
+    accessLevel: string,
+  ) {
```

Comment:
This are quite a lot function parameters. Would be better to use objects o group some of them? i.e. 

```suggestion
  private async createMySQLDatabase(
    p: ProcessRenderer,
    projectId: string,
    dbOptions: {...},
    userOptions: {...},
  ) {
```

Reply:
Good suggestion; I refactored this a bit.